app-backend/common/src/main/scala/com/rasterfoundry/common/color/ColorCorrect.scala

```diff
@@ -12,236 +11,156 @@ import geotrellis.raster.UByteConstantNoDataCellType
-          else if (z < newMin) newMin
-          else z.toInt
+          if (pixelValue > newMax) newMax
+          else if (pixelValue < newMin) newMin
+          else pixelValue.toInt
         }
 
-        gammaOpt match {
-          case None => v
-          case Some(gamma) =>
-            clampColor {
-              val gammaCorrection = 1 / gamma
-              (255 * FastMath.pow(v / 255.0, gammaCorrection)).toInt
-            }
-        }
+        v
       } else {
         val v = {
-          val scaled = (((z - oldMin) * dNew) / dOld) + newMin
+          val scaled = (((pixelValue - oldMin) * dNew) / dOld) + newMin
 
           if (scaled > newMax) newMax
           else if (scaled < newMin) newMin
           else scaled.toInt
         }
 
-        gammaOpt match {
-          case None => v
-          case Some(gamma) =>
-            clampColor {
-              val gammaCorrection = 1 / gamma
-              (255 * FastMath.pow(v / 255.0, gammaCorrection)).toInt
-            }
-        }
+        v
       }
-    } else z.toInt
+    } else pixelValue.toInt
   }
 
   val rgbBand: (Option[Int], Option[Int], Int) => Some[Int] =
     (specificBand: Option[Int], allBands: Option[Int], tileDefault: Int) =>
       specificBand.fold(allBands)(Some(_)).fold(Some(tileDefault))(x => Some(x))
 
-  def complexColorCorrect(rgbTile: MultibandTile, saturation: Saturation)(
-      layerNormalizeArgs: Map[Int, ClipBounds],
-      gammas: Map[Int, Option[Double]]
-  )(sigmoidalContrast: SigmoidalContrast)(
-      colorCorrectArgs: Map[Int, MaybeClipBounds],
-      tileClipping: MultiBandClipping,
-      noDataValue: Option[Double]
-  ): MultibandTile = {
+  def complexColorCorrect(rgbTile: MultibandTile,
+                          layerNormalizeArgs: Map[String, ClipBounds],
+                          noDataValue: Option[Double]): MultibandTile = {
     val (red, green, blue) = (rgbTile.band(0), rgbTile.band(1), rgbTile.band(2))
-    val (gr, gg, gb) = (gammas(0), gammas(1), gammas(2))
     val tileCellType = UByteConstantNoDataCellType
-    val (nred, ngreen, nblue) = (
+    val (dstRed, dstGreen, dstBlue) = (
       ArrayTile.alloc(tileCellType, rgbTile.cols, rgbTile.rows),
       ArrayTile.alloc(tileCellType, rgbTile.cols, rgbTile.rows),
       ArrayTile.alloc(tileCellType, rgbTile.cols, rgbTile.rows)
     )
 
-    val ClipBounds(rmin, rmax) = layerNormalizeArgs(0)
-    val ClipBounds(gmin, gmax) = layerNormalizeArgs(1)
-    val ClipBounds(bmin, bmax) = layerNormalizeArgs(2)
+    /** These are all super unsafe destructurings
+      * I tried to make the assumptions more explicit by naming the keys instead
+      * of numbering them, but if you see this and feel nervous about it, that's
+      * reasonable.
+      * It's safe, because the layerNormalizeArgs are constructed and consumed in
+      * this and only this file, so there's no way for this map to get all wonky,
+      * but, future reader, your concern is not unreasonable.
```

Comment:
past writer, your argument is convincing and this present reader feels that it is explicit enough